documentation/docs/04-hooks.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Hooks
+---
+
+An optional `src/hooks.js` (or `src/hooks.ts`, or `src/hooks/index.js`) file exports three functions, all optional, that run on the server — **getContext**, **getSession** and **handle**.
+
+> The location of this file can be [configured](#configuration) as `config.kit.files.hooks`
+
+### getContext
+
+This function runs on every incoming request. It generates the `context` object that is available to [endpoint handlers](#routing-endpoints) as `request.context`, and used to derive the [`session`](#hooks-getsession) object available in the browser.
+
+If unimplemented, context is `{}`.
+
+```ts
+type Incoming = {
```

Comment:
it's less work for us but a lot more work for users of the docs. medium term it would be nice if we were able to auto-generate these blocks, perhaps, but for now my inclination is to maintain them manually